pkg/controller/shared_state.go

```diff
@@ -79,21 +79,26 @@ type SharedState struct {
 	conditionWatcher              *watcher.ConditionWatcher
 	azureDiskCSITranslator        csitranslator.InTreePlugin
 	availableAttachmentsMap       sync.Map
+	crdProvisioner                CrdProvisioner
 }
 
-func NewSharedState(driverName, objectNamespace, topologyKey string, eventRecorder record.EventRecorder, cachedClient client.Client, azClient azdisk.Interface, kubeClient kubernetes.Interface, crdClient crdClientset.Interface) *SharedState {
+type CrdProvisioner interface {
```

Comment:
Consider `DriveLifecycle` to avoid confusion with `CrdProvisioner` defined elsewhere.